src/PolyType.Roslyn/Model/AssociatedTypeModel.cs

```diff
@@ -8,4 +8,33 @@ namespace PolyType.Roslyn;
 /// <param name="AssociatedType">The symbol for the related type.</param>
 /// <param name="AssociatingAssembly">The assembly that declared the association.</param>
 /// <param name="Location">The location where the association is declared.</param>
-public record struct AssociatedTypeModel(INamedTypeSymbol AssociatedType, IAssemblySymbol AssociatingAssembly, Location? Location);
+/// <param name="Requirements">The associated type's requirements.</param>
+public record struct AssociatedTypeModel(INamedTypeSymbol AssociatedType, IAssemblySymbol AssociatingAssembly, Location? Location, TypeShapeDepth Requirements);
+
+/// <summary>
+/// Describes the requirements for preparing an associated type.
+/// </summary>
+/// <devremarks>
+/// Keep this in sync with the TypeShapeDepth enum defined in the PolyType assembly.
+/// </devremarks>
+[Flags]
+public enum TypeShapeDepth
+{
+    /// <summary>No shape is required.</summary>
```

Comment:
I presume this would still result in an empty shape `ITypeShape<T>` for the type being generated, correct?
```suggestion
    /// <summary>Generates an empty <see cref="ITypeShape{T}" /> without any transitive shape information.</summary>
```

Reply:
Honestly I didn't anticipate anyone using this. It's just customary to define a 0 value on flags enums. I suppose I should test its effect though.

Reply:
The behavior _was_ that we wouldn't generate empty shapes. But based on your expectation, I've modified the code and added tests to verify empty shapes are generated.

Reply:
Sounds good. An empty shape is still useful since it can be used with an `ITypeShapeFunc` for example to extract results from a `ValueTask<T>`.